Explorer/Assets/DCL/Infrastructure/ECS/SceneLifeCycle/SceneLoadingLimit/SceneLoadingLimit.cs

```diff
@@ -0,0 +1,165 @@
+using Cysharp.Threading.Tasks;
+using DCL.Optimization.PerformanceBudgeting;
+using DCL.Utilities;
+using ECS.SceneLifeCycle.SceneDefinition;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using Utility;
+
+namespace ECS.SceneLifeCycle.IncreasingRadius
+{
+
+
+    public class SceneLoadingLimit
+    {
+
+        private readonly Dictionary<SceneLimitsKey, SceneLimits> constantSceneLimits = new ()
+        {
+            // 1 scene, 1 high quality LOD, 10 low quality LODs. Limit: 561MB
+            { SceneLimitsKey.LOW_MEMORY, new SceneLimits(SceneLoadingMemoryConstants.MAX_SCENE_SIZE + SceneLoadingMemoryConstants.MAX_SCENE_LOD, 10 * SceneLoadingMemoryConstants.MAX_SCENE_LOWQUALITY_LOD) },
+
+            // 3 scenes, 5 high quality LODs, 30 low quality LODs. Limit: 1925MB
+            { SceneLimitsKey.MEDIUM_MEMORY, new SceneLimits((3 * SceneLoadingMemoryConstants.MAX_SCENE_SIZE) + (5 * SceneLoadingMemoryConstants.MAX_SCENE_LOD), 30 * SceneLoadingMemoryConstants.MAX_SCENE_LOWQUALITY_LOD) },
+
+            // No limits.
+            { SceneLimitsKey.MAX_MEMORY, new SceneLimits(float.MaxValue, float.MaxValue) },
+
+            // 1 scene, 1 high quality LOD. Could be useful for debugging single scenes
+            { SceneLimitsKey.WARNING, new SceneLimits(1, 5 * SceneLoadingMemoryConstants.MAX_SCENE_LOWQUALITY_LOD) },
+        };
+
+
+        public SceneLimits currentSceneLimits { get; private set; }
+
+        private float SceneCurrentMemoryUsageInMB;
+        private float QualityReductedLODCurrentMemoryUsageInMB;
+        private SceneLimitsKey initialKey;
+        private SceneTransitionState sceneTransitionState;
```

Comment:
It looks like (unless I am mistaken) `SceneTransitionState` exists only to manipulate the state of the async process. If we get rid of the latter, won't it be possible to get rid of the state as well? Can you evaluate it?

Or if you could actually merge `isMemoryNormal` and `isAbundance` to a single `enum` you could store that value in this class to determine when the direction of the interpolation inverses.